File: sources/api/Models/Expense.cs
using System.ComponentModel.DataAnnotations;

namespace DotNetAPI.Model
{
    public class Expense
    {
        public int Id { get; set; }
        public required int UserId { get; set; }
        public required int GroupId { get; set; }
        public required ICollection<int> UserIdInvolved { get; set; } // Collection of user IDs involved in the expense
        public required int CategoryId { get; set; }
        public required float Amount { get; set; }
        public required int Date { get; set; }
        public required string Place { get; set; } // Enum ??
        public string? Description { get; set; }
    }
}

File: sources/mobile/app/src/main/java/com/console/ratcord/domain/entity/expense/Expense.kt
package com.console.ratcord.domain.entity.expense

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey
import com.console.ratcord.domain.entity.category.Category
import com.console.ratcord.domain.entity.group.Group
import com.console.ratcord.domain.entity.user.User
import com.console.ratcord.domain.entity.user.UserExtraMinimal
import com.console.ratcord.domain.entity.user.UserMinimal
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
@Entity(
    tableName = "Expense",
    foreignKeys = [
        ForeignKey(
            entity = User::class,
            parentColumns = ["id"],
            childColumns = ["userId"],
            onDelete = ForeignKey.CASCADE
        ),
        ForeignKey(
            entity = Group::class,
            parentColumns = ["id"],
            childColumns = ["groupId"],
            onDelete = ForeignKey.CASCADE
        ),
        ForeignKey(
            entity = Category::class,
            parentColumns = ["id"],
            childColumns = ["categoryId"],
            onDelete = ForeignKey.CASCADE
        ),
        ForeignKey(
            entity = User::class,
            parentColumns = ["id"],
            childColumns = ["userIdInvolved"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class Expense(
    @PrimaryKey val id: Int,
    val user: UserExtraMinimal,
    val groupId: Int,
    val usersInvolved: List<UserExtraMinimal>,
    val category: Category,
    val amount: Float,
    val date: Int,
    val place: String,
    val description: String,
    val image: String?
)